src/m365/viva/commands/engage/engage-community-add.spec.ts

```diff
@@ -0,0 +1,329 @@
+        privacy: "public"
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when invalid privacy option is provided', async () => {
+    const actual = await command.validate({
+      options: {
+        displayName: "Software engineers",
+        description: "A community for all software engineer",
+        privacy: "invalid"
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when invalid adminEntraId is provided', async () => {
+    const actual = await command.validate({
+      options: {
+        displayName: "Software engineers",
+        description: "A community for all software engineer",
+        privacy: "private",
+        adminEntraIds: "invalid"
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation when invalid adminEntraUserName is provided', async () => {
+    const actual = await command.validate({
+      options: {
+        displayName: "Software engineers",
+        description: "A community for all software engineer",
+        privacy: "private",
+        adminEntraUserNames: "invalid"
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if both adminEntraIds and adminEntraUserNames are specified', async () => {
+    const actual = await command.validate({
+      options: {
+        displayName: "Software engineers",
+        description: "A community for all software engineer.",
+        privacy: "public",
+        adminEntraIds: "50674d84-6bf1-470b-89b5-d55ce0a5a720",
+        adminEntraUserNames: "john.doe@contoso.onmicrosoft.com"
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation when valid options are provided with adminEntraIds', async () => {
+    const actual = await command.validate({
+      options: {
+        displayName: "Software engineers",
+        description: "A community for all software engineer.",
+        privacy: "public",
+        adminEntraIds: "50674d84-6bf1-470b-89b5-d55ce0a5a720"
+      }
+    }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when valid options are provided with adminEntraUserNames', async () => {
+    const actual = await command.validate({
+      options: {
+        displayName: "Software engineers",
+        description: "A community for all software engineer.",
+        privacy: "public",
+        adminEntraUserNames: "john.doe@contoso.onmicrosoft.com"
+      }
+    }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('creates a community without waiting for provisioning to complete', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts: any) => {
```

Comment:
Options object is in fact already typed. Let's not cast it to anonymous. This applies to the other tests as well.

```suggestion
    sinon.stub(request, 'post').callsFake(async (opts) => {
```